### PR TITLE
[stable/prometheus-operator] Add prometheus-operator ServiceAccount annotations

### PR DESCRIPTION
**Pull request by koziez**
Signed-off-by: Max Roycroft Jr. <max.roycroft.jr@gmail.com>

#### Is this a new chart
No, this enhances the stable/prometheus-operator.

#### What this PR does / why we need it:
Add missing support for prometheus-operator ServiceAccount annotations.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by vsliouniaev** [comment]
/ok-to-test
Looks like you need to add sign-off to your commits

File: stable/prometheus-operator/templates/prometheus-operator/serviceaccount.yaml [hunk 1]
```diff
@@ -6,6 +6,10 @@ metadata:
   namespace: {{ template "prometheus-operator.namespace" . }}
   labels:
     app: {{ template "prometheus-operator.name" . }}-operator
+{{- if .Values.prometheusOperator.serviceAccount.annotations }}s
```
Comment (mrueg) [inline stable/prometheus-operator/templates/prometheus-operator/serviceaccount.yaml]: the s at the end should probably be removed?

File: stable/prometheus-operator/values.yaml [hunk 2]
```diff
@@ -1184,6 +1184,7 @@ prometheusOperator:
   serviceAccount:
     create: true
     name: ""
+    annotation: {}
```
Comment (mrueg) [inline stable/prometheus-operator/values.yaml]: One is using annotation, the other one annotations. Can we use the plural everywhere?

Comment (koziez) [reply stable/prometheus-operator/values.yaml]: Sorry for the delay. Has been fixed.

File: stable/prometheus-operator/templates/prometheus-operator/serviceaccount.yaml (hunk 1 above)
Comment (koziez) [reply stable/prometheus-operator/templates/prometheus-operator/serviceaccount.yaml]: Fixed values.yaml to make consistent.

File: stable/prometheus-operator/values.yaml (hunk 2 above)
Comment (koziez) [reply stable/prometheus-operator/values.yaml]: Fixed.

File: stable/prometheus-operator/templates/prometheus-operator/serviceaccount.yaml (hunk 1 above)
Comment (mrueg) [inline stable/prometheus-operator/templates/prometheus-operator/serviceaccount.yaml]:
```suggestion
{{- if .Values.prometheusOperator.serviceAccount.annotations }}
```

File: stable/prometheus-operator/Chart.yaml [hunk 3]
```diff
@@ -12,7 +12,7 @@ sources:
   - https://github.com/coreos/kube-prometheus
   - https://github.com/coreos/prometheus-operator
   - https://coreos.com/operators/prometheus
-version: 9.3.0
+version: 9.3.1
```
Comment (mrueg) [inline stable/prometheus-operator/Chart.yaml]: this needs to be bumped to 9.3.2

**Comment by scottrigby** [comment]
/hold

Let’s hold PRs to stable/prometheus-operator chart for a short time until https://github.com/prometheus-community/community/issues/28#issuecomment-672933972 is resolved. If you want to help with that effort, to test etc, that would be great!

**Comment by scottrigby** [comment]
🚧 This chart is now deprecated.

Further development has moved to [prometheus-community/helm-charts](https://github.com/prometheus-community/helm-charts). The chart has been renamed [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) to more clearly reflect that it installs the `kube-prometheus` project stack, within which Prometheus Operator is only one component.

If applicable, please re-open this PR there. Thanks!

**Pull request closed without merging**